appstudio-utils/util-scripts/lib/fetch/git.sh

```diff
@@ -5,21 +5,28 @@ git-fetch-repo() {
   local directory="$3"
 
   mkdir -p "$directory"
-  cd "$directory"
+  cd "$directory" || return
 
   # Git clone might work but this avoids downloading the entire repo
-  git init -q .
-  git fetch -q --depth 1 --no-tags $repo $ref_or_sha
-  git checkout -q FETCH_HEAD
+  git init -q . && \
```

Comment:
Hmm why is this needed? `set -e` is there so we don't have to do this, right?

Reply:
I wanted to prevent any output to stdout so that the `$sha` below is not set to any value. Without this tests fail with:

```
  1) git-fetch-repo fails on bogus repository
     When call git-fetch-repo bogus main /tmp/tmp.feHVReVNzS

     1.1) WARNING: There was output to stdout but not found expectation

            stdout:FETCH_HEAD
            

          # test/fetch-git_spec.sh:64-69

  2) git-fetch-repo fails on bogus ref
     When call git-fetch-repo /tmp/tmp.f1kpx6bRwP bogus /tmp/tmp.O5JknTWyrf

     2.1) WARNING: There was output to stdout but not found expectation

            stdout:FETCH_HEAD
            

          # test/fetch-git_spec.sh:71-76
```

Which tells me that these commands progress regardless of `set -e`.